05_ANALYSIS_TOOLS/etl_pipeline/transform/business_metrics.py

```
@@ -0,0 +1,637 @@
+                if monthly_revenue > 0:
+                    monthly_ebitda = monthly_revenue - monthly_operational_expenses
+                    monthly_ebitdas.append(monthly_ebitda)
+                    logger.info(f"Monthly EBITDA for {pnl_key}: ${monthly_ebitda:,.2f} (Revenue: ${monthly_revenue:,.2f}, Op Expenses: ${monthly_operational_expenses:,.2f}, Total Expenses: ${monthly_total_expenses:,.2f})")
+            
+            if monthly_ebitdas:
+                # Calculate average monthly EBITDA
+                avg_monthly_ebitda = sum(monthly_ebitdas) / len(monthly_ebitdas)
+                logger.info(f"Average monthly EBITDA (company-wide): ${avg_monthly_ebitda:,.2f} (from {len(monthly_ebitdas)} months)")
+                
+                # Apply location adjustment factor to get EBITDA for sale locations only
+                adjusted_monthly_ebitda = avg_monthly_ebitda * location_adjustment_factor
```

Ruff (0.12.2)

Undefined name `location_adjustment_factor`

(F821)